### PR TITLE
executor: add test for rebuilding prepared statement plan.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by tiancaiamao** [comment]
LGTM

File: executor/prepared_test.go [hunk 1]
```diff
@@ -76,6 +77,16 @@ func (s *testSuite) TestPrepared(c *C) {
 		c.Assert(err, IsNil)
 		c.Assert(stmt.OriginText(), Equals, query)
 
+		// Check that rebuild plan works.
+		tk.Se.PrepareTxnCtx(goctx.Background())
+		err = stmt.RebuildPlan()
+		c.Assert(err, IsNil)
+		rs, err := stmt.Exec(goctx.Background())
+		c.Assert(err, IsNil)
+		_, err = rs.Next()
```
Comment (shenli) [inline executor/prepared_test.go]: Should we check the return value?

**Comment by tiancaiamao** [comment]
B.T.W, can you uncomment here https://github.com/pingcap/tidb/blob/master/new_session_test.go#L1446
@coocood

**Comment by coocood** [comment]
@shenli 
SInce we rebuild the plan in the middle, we can't use testkit to check value, the test code would be tedious, but the nil error already confirmmed the rebuild plan works.

**Pull request merged**